lib/p2p/Pool.ts

```diff
@@ -0,0 +1,172 @@
+import net, { Server, Socket } from 'net';
+import { EventEmitter } from 'events';
+import Peer from './Peer';
+import Hosts from './Hosts';
+import NetAddress from './NetAddress';
+import Logger from '../Logger';
+
+interface PoolConfig {
+  listen: boolean;
+  port: number;
+}
+
+/** A pool of peers for handling all network activity */
+class Pool extends EventEmitter {
```

Comment:
Yes, `Pool` is a quite ambiguous, but since other options seems be too verbose, and `Pool` is under `p2p` directory, I thought it can work.
I think it should either be `Pool` or `PeersPool`. Let me know what you prefer.

Reply:
Thanks, good point that it's under `p2p`. Lets just keep it as you have it here. One thing I was thinking about is eventually using `index.ts` files in folders with lots of files/classes, such that one day we might do `import { Peer } from './p2p'` which is not ambiguous at all.